cmd/generate-workflow.go

```diff
@@ -74,44 +57,67 @@ func init() {
 	rootCmd.AddCommand(newGenerateWorkflowCmd())
 }
 
-func (gwc *generateWorkflowCmd) generateWorkflows(dest string, deployType string, flagVariables []string, templateWriter templatewriter.TemplateWriter, flagValuesMap map[string]string) error {
-	if flagValuesMap == nil {
-		return fmt.Errorf("flagValuesMap is nil")
-	}
+func (gwc *generateWorkflowCmd) generateWorkflows() error {
 	var err error
-	for _, flagVar := range flagVariables {
-		flagVarName, flagVarValue, ok := strings.Cut(flagVar, "=")
-		if !ok {
-			return fmt.Errorf("invalid variable format: %s", flagVar)
-		}
-		flagValuesMap[flagVarName] = flagVarValue
-		log.Debugf("flag variable %s=%s", flagVarName, flagVarValue)
-	}
 
-	if deployType == "" {
-		selection := &promptui.Select{
-			Label: "Select k8s Deployment Type",
-			Items: []string{"helm", "kustomize", "manifests"},
-		}
+	flagVariablesMap = flagVariablesToMap(gwc.flagVariables)
 
-		_, deployType, err = selection.Run()
-		if err != nil {
-			return err
+	if gwc.deployType == "" {
+		if flagValue := flagVariablesMap["deploy-type"]; flagValue == "helm" || flagValue == "kustomize" || flagValue == "manifests" {
+			gwc.deployType = flagVariablesMap["deploy-type"]
+		} else {
+			selection := &promptui.Select{
+				Label: "Select k8s Deployment Type",
+				Items: []string{"helm", "kustomize", "manifests"},
+			}
+
+			_, gwc.deployType, err = selection.Run()
+			if err != nil {
+				return err
+			}
 		}
 	}
 
-	workflow := workflows.CreateWorkflowsFromEmbedFS(template.Workflows, dest)
-	workflowConfig, err := workflow.GetConfig(deployType)
+	workflow := workflows.CreateWorkflowsFromEmbedFS(template.Workflows, gwc.dest)
+	draftConfig, err := workflow.GetConfig(gwc.deployType)
 	if err != nil {
 		return fmt.Errorf("get config: %w", err)
 	}
 
-	customInputs, err := prompts.RunPromptsFromConfigWithSkips(workflowConfig, maps.Keys(flagValuesMap))
-	if err != nil {
+	handleFlagVariables(flagVariablesMap, draftConfig)
+
+	if err = prompts.RunPromptsFromConfigWithSkips(draftConfig); err != nil {
 		return err
 	}
 
-	maps.Copy(customInputs, flagValuesMap)
+	if err := workflows.UpdateProductionDeployments(gwc.deployType, gwc.dest, draftConfig, gwc.templateWriter); err != nil {
+		return fmt.Errorf("update production deployments: %w", err)
+	}
+
+	return workflow.CreateWorkflowFiles(gwc.deployType, draftConfig, gwc.templateWriter)
+}
+
+func handleFlagVariables(flagVariablesMap map[string]string, draftConfig *config.DraftConfig) {
+	for flagName, flagValue := range flagVariablesMap {
+		log.Debugf("flag variable %s=%s", flagName, flagValue)
+		// handles flags that are meant to represent environment arguments
+		if variable, err := draftConfig.GetVariable(flagName); err != nil {
+			log.Infof("adding new environment argument %s", flagName)
+			draftConfig.AddVariable(flagName, flagValue)
+		} else {
```

Comment:
this if/else can be bypassed by using something like a PutVariable method, and debug logging variable creation from within it